src/PosixSemaphore.php

```diff
@@ -232,21 +228,42 @@ private function open(): void
      */
     private function init(int $maxLocks, int $permissions): void
     {
-        if (\msg_queue_exists($this->key)) {
-            throw new SyncException('A semaphore with that ID already exists');
+        if (!self::$nextId) {
+            self::$nextId = \random_int(1, \PHP_INT_MAX - 1);
         }
 
-        /** @psalm-suppress TypeDoesNotContainType */
-        $queue = \msg_get_queue($this->key, $permissions);
-
-        /** @psalm-suppress TypeDoesNotContainType */
-        if (!$queue) {
-            throw new SyncException('Failed to create the semaphore.');
+        \set_error_handler(static function (int $errno, string $errstr): bool {
+            static $attempts = 0;
+            if (++$attempts > 10) {
+                throw new SyncException('Failed to create semaphore: ' . $errstr, $errno);
+            }
+            return true;
+        });
+
+        try {
+            do {
+                while (\msg_queue_exists($id = self::$nextId)) {
+                    self::$nextId = self::$nextId % \PHP_INT_MAX + 1;
```

Comment:
This likely needs to be restricted to 32 bit?

Reply:
This worked fine on macOS, but for some systems you're probably right.